staging/src/k8s.io/kms/encryption/aesgcm.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryption
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/rand"
+	"errors"
+	"sync/atomic"
+	"time"
+
+	"k8s.io/apiserver/pkg/storage/value"
+	kaes "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+)
+
+const (
+	// safeUsage with 2^21 is a very defensive value.  2^21 is calculated by
+	// assuming a probability of 1/2^80 for the birthday problem.
+	safeUsage uint32 = 1 << 21
+	// maxUsage is closer to the recommended maximum usage of 2^32 times.
+	maxUsage uint32 = 1 << 31
+	// spaceUsage is a defensive maximum amount of bytes of data that should be
+	// encrypted with a key and a nonce. It is approximately 32 GiByte.
+	// The maximum usage is 2^36 - 256 byte, which is approximately 64 GiByte.
+	spaceUsage uint64 = 1 << 35
+
+	// keySize is the key size in bytes
+	keySize = 128 / 8
+)
+
+var (
+	// week was picked by during discussions of KMSv2 development.
+	week = time.Hour * 24 * 7
+	skew = time.Minute
```

Comment:
Not sure why we have this?

Reply:
I added this in certain parts of the code, where I would like to ensure that we are definitely within the time frame we want.

Removed it.